src/particles/elements/Aperture.H

```diff
@@ -104,8 +108,19 @@ namespace impactx
             shift_in(x, y, px, py);
 
             // scale horizontal and vertical coordinates
-            amrex::ParticleReal const u = x / m_xmax;
-            amrex::ParticleReal const v = y / m_ymax;
+            amrex::ParticleReal u;
+            amrex::ParticleReal v;
+            amrex::ParticleReal dx = m_xmax;
+            amrex::ParticleReal dy = m_ymax;
+            // first case: a single aperture
+            if (m_repeat_x == 0.0 || m_repeat_y == 0.0) {
```

Comment:
How does a periodic slit then work? Just by making the apertures touch each other in the direction of the slit? As a user, I would probably intuitively define only repeat_x and define an elongated rectangular aperture in y. 

 So maybe this could be an (m_repeat_x == 0.0 **&&** m_repeat_y == 0.0) ?
And then the slit can be done via an option: (m_repeat_x == 0.0 **||** m_repeat_y == 0.0)